go/marshal/decode_test.go

```diff
@@ -549,3 +549,42 @@ func TestDecodeMapWrongNomsType(t *testing.T) {
 	var m map[string]int
 	assertDecodeErrorMessage(t, types.NewList(types.String("a"), types.Number(1)), &m, "Cannot unmarshal List<Number | String> into Go value of type map[string]int")
 }
+
+func TestDecodeOntoInterface(t *testing.T) {
+	assert := assert.New(t)
+
+	var i interface{}
+	err := Unmarshal(types.Number(1), &i)
+	assert.NoError(err)
+	assert.Equal(float64(1), i)
+
+	err = Unmarshal(types.String("abc"), &i)
+	assert.NoError(err)
+	assert.Equal("abc", i)
+
+	err = Unmarshal(types.Bool(true), &i)
+	assert.NoError(err)
+	assert.Equal(true, i)
+
+	err = Unmarshal(types.NewList(types.String("abc")), &i)
+	assert.NoError(err)
+	assert.Equal([]interface{}{"abc"}, i)
```

Comment:
We could generate `[]string` in this case. Is it worth it? The static type will always be `interface{}` or `[]interface` but I guess it would lead to less type casts.